index.js

```diff
@@ -27,8 +27,16 @@ module.exports = function conversation({name, app, appId,
   requestId = 'EdwRequestId.33ac9138-640f-4e6e-ab71-b9619b2c2210',
   locale = 'en-US',
   fixSpaces = false,
-  fuzzyDistance = 0.93
+  fuzzyDistance = 0.93,
+  handler = null
```

Comment:
As params and default params are available to later default params, I think you should be able to write something like this directly inline:

```js
function conversation({name, app, appId
    //...
    handler = app.handler || null
}){
    if(handler === null) throw new Error("Must provide either an app or handler.");
    //...
};
```
In the above code, if `handler` is passed it will still take precedence which is what we want. Only if handler is not passed and app.handler is falsy, it will take null as value.

It seems cleaner and more concise to me, as it avoids creating an anonymous function which might never be executed and could be confusing.